Shallow Rendering

tamarau opened the pull request:
Hi, 

My first attempt to translate shallow rending. 

I've got a couple of things I'm not too sure about so would like to ask for feedback here :)
- I called shallow rendering 'Flaches Rendern'; rendern because I don't think there is a good German equivalent we use? I've seen 'Darstellung' but it's not what I would use when talking to other devs about rendering.
- Not sure if I can leave 'unit tests' and 'high-level API' as they are? I marked them in italics for now until that is solved. :)
- General question: how does the side menu get updated to the new title of the page?

Thanks ☺️

ph1p commented on content/docs/addons-shallow-renderer.md:
```diff
@@ -1,68 +1,67 @@
 ### `shallowRenderer.render()` {#shallowrendererrender}
 
-You can think of the shallowRenderer as a "place" to render the component you're testing, and from which you can extract the component's output.
+Man kann sich shallowRenderer als den "Ort" vorstellen, in der man die Komponente testet, und von wo aus der Output der Komponente entnommen wird.
```
```suggestion
Du kannst dir den shallowRenderer als den "Ort" vorstellen an dem die zu testende Komponente gerendert wird, und du daraus die Ausgabe der Komponente entnehmen kannst.
```

ph1p commented on content/docs/addons-shallow-renderer.md:
```diff
@@ -1,68 +1,67 @@
+Man kann sich shallowRenderer als den "Ort" vorstellen, in der man die Komponente testet, und von wo aus der Output der Komponente entnommen wird.
 
-`shallowRenderer.render()` is similar to [`ReactDOM.render()`](/docs/react-dom.html#render) but it doesn't require DOM and only renders a single level deep. This means you can test components isolated from how their children are implemented.
+`shallowRenderer.render()` ist ähnlich wie [`ReactDOM.render()`](/docs/react-dom.html#render), nur dass es keinen DOM benötigt und nur einen Level tief rendert. Das bedeutet, dass man Komponenten isoliert von der Art und Weise wie deren Kinder-Komponenten implementiert werden, testen kann.
```
```suggestion
`shallowRenderer.render()` ist ähnlich wie [`ReactDOM.render()`](/docs/react-dom.html#render), nur dass es kein DOM benötigt und nur einen Level tief rendert. Das bedeutet, dass du Komponenten abgegrenzt testen kannst, unabhängig davon wie die Kind-Komponenten implementiert sind.
```

ph1p commented on content/docs/addons-shallow-renderer.md:
```diff
@@ -1,68 +1,67 @@
 ### `shallowRenderer.getRenderOutput()` {#shallowrenderergetrenderoutput}
 
-After `shallowRenderer.render()` has been called, you can use `shallowRenderer.getRenderOutput()` to get the shallowly rendered output.
+Nachdem `shallowRenderer.render()` aufgerufen worden ist, bekommt man mit `shallowRenderer.getRenderOutput()` den flachen gerenderten Output.
```
```suggestion
Nachdem `shallowRenderer.render()` aufgerufen wurde, kannst du dir mit `shallowRenderer.getRenderOutput()` die flach gerenderte Ausgabe zurückgeben lassen.
```

ph1p commented:
> Hi,
> 
> My first attempt to translate shallow rending.
> 
> I've got a couple of things I'm not too sure about so would like to ask for feedback here :)
> 
> * I called shallow rendering 'Flaches Rendern'; rendern because I don't think there is a good German equivalent we use? I've seen 'Darstellung' but it's not what I would use when talking to other devs about rendering.

Hi Tamara!

First. Thanks for your PR ;)

Right now I think "Flaches Rendern" is ok. You have also added the English words so that everyone can understand them. 

> * Not sure if I can leave 'unit tests' and 'high-level API' as they are? I marked them in italics for now until that is solved. :)

I changed "high level" to "übergeordnet". Might sound a bit better. Let me know what you think. "Unit Tests" is fine.

> * General question: how does the side menu get updated to the new title of the page?

Just update the `nav.yml`

tamarau commented on content/docs/addons-shallow-renderer.md:
````diff
@@ -1,68 +1,67 @@
 ```
 
-Shallow testing currently has some limitations, namely not supporting refs.
+Zurzeit hat flaches Rendering die Einschränkung, dass `refs` nicht unterstützt werden.
````
Changed it a little bit in my commit :)

ph1p replied:
You translate `Shallow testing` into `flaches Rendering`. Why?

ph1p replied:
I have read the docs again and it makes no difference here between testing and rendering. So your translation doesn't lose any sense, so I think it's good. ;)

tamarau commented:
Hi Phil, 

Thanks for your feedback! Really appreciated :) 

I've committed your changes and updated nav.yml. I also changed 'Output' in the last line to 'Ausgabe', so it matches the previous sentence.
